docs/data/data-grid/aggregation/aggregation.md

```diff
@@ -4,15 +4,277 @@ title: Data Grid - Aggregation
+The aggregation can be modified through the grid interface by opening the column menu and selecting an item on the _Aggregation_ select.
 
-When grouping, you will be able to apply an aggregation function to populate the group row with values.
+The aggregated values will be rendered in a footer row at the bottom of the grid.
+
+:::warning
+⚠️ The footer row will be pinned at the bottom of the grid once [#1251](https://github.com/mui/mui-x/issues/1251) is ready.
+:::
+
+{{"demo": "AggregationBasic.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Pass aggregation to the grid
+
+### Structure of the model
+
+The aggregation model is a lookup where the keys are the field of the columns to aggregate and the values are aggregation items describing how those columns should be aggregated.
+
+An aggregation item is an object where `item.footer` contains the name of the aggregation function to apply on the footers
+and `item.inline` contains the name of the aggregation function to apply on the grouping rows.
+
+:::info
+If you only want to aggregate on the footer, you can use the short notation
+
+```ts
+const model: GridAggregationModel = { gross: { footer: 'sum' } };
+// can be replaced by
+const model: GridAggregationModel = { gross: 'sum' };
+```
+
+:::
+
+### Initialize the aggregation
+
+To initialize the aggregation without controlling it, provide the model to the `initialState` prop.
+
+{{"demo": "AggregationInitialState.js", "bg": "inline"}}
+
+### Controlled aggregation
+
+Use the `filterModel` prop to control the aggregation passed to the grid.
+
+You can use the `onFilterModelChange` prop to listen to changes to the filters and update the prop accordingly.
+
+{{"demo": "AggregationControlled.js", "bg": "inline"}}
+
+## Disable the aggregation
+
+### For all columns
+
+You can disable the aggregation by setting the `disableAggregation` prop to true.
+
+It will disable all features related to the aggregation, even if a model is provided.
+
+{{"demo": "AggregationDisabled.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### For some columns
+
+In case you need to disable the aggregation on specific column(s), set the `aggregatable` property on the respective column definition (`GridColDef`) to `false`.
+In the example below, the `title` and `year` columns can not be aggregated.
+
+{{"demo": "AggregationColDefAggregable.js", "bg": "inline", "defaultCodeOpen": false}}
+
+## Aggregation label
+
+### Choose the column for the label
+
+Use the `aggregationFooterLabelField` to choose on which column the label of the aggregation should be rendered.
+
+{{"demo": "AggregationLabelColumn.js", "bg": "inline"}}
+
+### Customize the content of the label
+
+TODO
+
+## Usage with row grouping
+
+When the row grouping is enabled, the aggregated values can be displayed on two positions:
+
+1. `footer` - the grid will add a root level footer to aggregate all the rows and one footer per group to aggregate its rows
```

Comment:
It is `footer`

You don't have `footer` on one side and `groupFooter` on this other
You have `footer`, and you can play with `isGroupAggregated` if you don't want to aggregate on one or the other

The only limitation is that you can't put an aggregation function on the group footers and another one on the root group footer.
But I don't think this use case it worth the additional complexity.
Because if we go that way, we could have people wanting to aggregate based on different functions depending on the grouping criteria (for instance when grouping by `["company", "director"]`, group the gross by  `max` on company footers and `sum` on director footers.

So I would rather wait for returns of weird use cases and perhaps introduce a variation of the model with more granularity.

Reply:
@joserodolfofreitas

Reply:
related to: https://github.com/mui/mui-x/pull/4208/files#r886680451